ciao-launcher/main.go

```diff
@@ -390,50 +389,51 @@ func connectToServer(doneCh chan struct{}, statusCh chan struct{}) {
 		dialCh <- err
 	}()
 
-	dialing := true
+	select {
+	case err := <-dialCh:
+		if err != nil {
+			break
+		}
+		clusterConfig, err := client.conn.ClusterConfiguration()
+		if err != nil {
+			glog.Errorf("Unable to get Cluster Configuration %v", err)
+			client.conn.Close()
+			break
+		}
+		computeNet = clusterConfig.Configure.Launcher.ComputeNetwork
+		mgmtNet = clusterConfig.Configure.Launcher.ManagementNetwork
+		diskLimit = clusterConfig.Configure.Launcher.DiskLimit
+		memLimit = clusterConfig.Configure.Launcher.MemoryLimit
+		if secretPath == "" {
+			secretPath = clusterConfig.Configure.Storage.SecretPath
+		}
+		if cephID == "" {
+			cephID = clusterConfig.Configure.Storage.CephID
+		}
+		printClusterConfig()
 
-DONE:
-	for {
-		select {
-		case err := <-dialCh:
-			dialing = false
-			if err != nil {
-				break DONE
-			}
-			clusterConfig, err := client.conn.ClusterConfiguration()
-			if err != nil {
-				glog.Errorf("Unable to get Cluster Configuration %v", err)
-				client.conn.Close()
-				break DONE
-			}
-			computeNet = clusterConfig.Configure.Launcher.ComputeNetwork
-			mgmtNet = clusterConfig.Configure.Launcher.ManagementNetwork
-			diskLimit = clusterConfig.Configure.Launcher.DiskLimit
-			memLimit = clusterConfig.Configure.Launcher.MemoryLimit
-			if secretPath == "" {
-				secretPath = clusterConfig.Configure.Storage.SecretPath
-			}
-			if cephID == "" {
-				cephID = clusterConfig.Configure.Storage.CephID
-			}
-			printClusterConfig()
+		client.installLauncherDeps()
 
-			client.installLauncherDeps()
+		err = startNetwork(doneCh)
+		if err != nil {
+			glog.Errorf("Failed to start network: %v\n", err)
+			client.conn.Close()
+			break
+		}
+		defer shutdownNetwork()
 
-			err = startNetwork(doneCh)
-			if err != nil {
-				glog.Errorf("Failed to start network: %v\n", err)
-				client.conn.Close()
-				break DONE
-			}
-			defer shutdownNetwork()
+		ovsCh = startOverseer(&wg, client)
+	case <-doneCh:
+		client.conn.Close()
+		<-dialCh
```

Comment:
I think we need a return here.  Otherwise we'll carry on down to the for loop and possibly call client.conn.Close again.  I'm not sure whether it's okay to do this or not.  Returning here should be safe as the overseer has not yet started so we don't need to wait on the waitgroup.  Also by returning here we do not print out the log overseer has exitted, which in this case is not strictly correct as it never started.